### PR TITLE
Deleted legacy CNV pipeline.

### PR DESCRIPTION
**Pull request by samuelklee**
Comprises the last ~~6 commits~~ 7 commits.  (Had to fix a test file dependency.  From now on, I'd like to request more encapsulation of test resources.  Certainly we should have a common pool of general, rarely changed resources, but sharing of specific resources across packages breaks encapsulation.)

@lbergelson I removed a few R dependencies.  We should update the base Docker image accordingly and make sure I didn't break anything.
@davidbenjamin I had to change one use of HashedListTargetCollection in CalculateContamination.  Also note that FilterByOrientationBias is the sole survivor in the exome package, so you may want to move it somewhere else.
@LeeTL1220 @vruano This removes a lot of your code.  Please speak up if there are any utility classes, etc. that you'd like to keep.  (For example, I kept the HMM code.)  I removed the Target codec and associated classes.
@sooheelee I will update the list of tools for doc updates accordingly.

### CODE REVIEW

**Comment by sooheelee** [comment]
Thank you @samuelklee. You are the best.

**Comment by davidbenjamin** [comment]
Wow, `OverlapDetector`.  Who knew?

**Comment by samuelklee** [comment]
@lbergelson @vruano @LeeTL1220 Would appreciate a quick review when you guys get a chance.  (Want to avoid people wasting time on kebab-case updates to or conflicts with this code.)

**Comment by samuelklee** [comment]
@LeeTL1220 Thanks for reviewing, but I think you might have commented on commits that were not part of this PR (which was just concerned with deleting files).  I'll still address your comments here, though.

File: scripts/cnv_cromwell_tests/somatic/run_cnv_somatic_workflows.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+sed -r "s/__GATK_DOCKER__/broadinstitute\/gatk\:$HASH_TO_USE/g" ${CNV_CROMWELL_TEST_DIR}/cnv_somatic_panel_wes_do-gc_workflow.json >cnv_somatic_panel_wes_do-gc_workflow_mod.json
+sed -r "s/__GATK_DOCKER__/broadinstitute\/gatk\:$HASH_TO_USE/g" ${CNV_CROMWELL_TEST_DIR}/cnv_somatic_panel_wgs_do-gc_workflow.json >cnv_somatic_panel_wgs_do-gc_workflow_mod.json
+sed -r "s/__GATK_DOCKER__/broadinstitute\/gatk\:$HASH_TO_USE/g" ${CNV_CROMWELL_TEST_DIR}/cnv_somatic_pair_wes_no-gc_workflow.json >cnv_somatic_pair_wes_no-gc_workflow_mod.json
+sed -r "s/__GATK_DOCKER__/broadinstitute\/gatk\:$HASH_TO_USE/g" ${CNV_CROMWELL_TEST_DIR}/cnv_somatic_pair_wgs_no-gc_workflow.json >cnv_somatic_pair_wgs_no-gc_workflow_mod.json
+sed -r "s/__GATK_DOCKER__/broadinstitute\/gatk\:$HASH_TO_USE/g" ${CNV_CROMWELL_TEST_DIR}/cnv_somatic_pair_wes_do-gc_workflow.json >cnv_somatic_pair_wes_do-gc_workflow_mod.json
+sed -r "s/__GATK_DOCKER__/broadinstitute\/gatk\:$HASH_TO_USE/g" ${CNV_CROMWELL_TEST_DIR}/cnv_somatic_pair_wgs_do-gc_workflow.json >cnv_somatic_pair_wgs_do-gc_workflow_mod.json
+
+echo "Running ========"
+CROMWELL_JAR="cromwell-0.28.jar"
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply scripts/cnv_cromwell_tests/somatic/run_cnv_somatic_workflows.sh]: Filed #3960.

File: scripts/cnv_wdl/somatic/cnv_somatic_pair_workflow.wdl [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+            entity_id = CollectCountsNormal.entity_id,
+            denoised_copy_ratios = DenoiseReadCountsNormal.denoised_copy_ratios,
+            het_allelic_counts = ModelSegmentsNormal.het_allelic_counts,
+            modeled_segments = ModelSegmentsNormal.modeled_segments,
+            ref_fasta_dict = ref_fasta_dict,
+            gatk4_jar_override = gatk4_jar_override,
+            gatk_docker = gatk_docker,
+            disk_space_gb = plot_normal_disk
+    }
+}
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply scripts/cnv_wdl/somatic/cnv_somatic_pair_workflow.wdl]: Added to #3940.

File: scripts/cnv_wdl/somatic/cnv_somatic_panel_workflow.wdl [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# Workflow for creating a GATK CNV Panel of Normals given a list of normal samples. Supports both WGS and WES.
+#
+# Notes:
+#
+# - Input file (normal_bams_list) must contain file paths to bam and bam index files separated by tabs in the following format:
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply scripts/cnv_wdl/somatic/cnv_somatic_panel_workflow.wdl]: I think you already filed #3460.  Will make a note there.

File: src/main/java/org/broadinstitute/hellbender/tools/copynumber/ModelSegments.java [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+                    "If multidimensional segmentation is performed, the total kernel used will be K_CR + S * K_AF.",
+            fullName = KERNEL_SCALING_ALLELE_FRACTION_LONG_NAME,
+            shortName = KERNEL_SCALING_ALLELE_FRACTION_SHORT_NAME,
+            minValue = 0.,
+            optional = true
+    )
+    private double kernelScalingAlleleFraction = 1.0;
+
+    @Argument(
+            doc = "Dimension of the kernel approximation.  A subsample containing this number of data points " +
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply src/main/java/org/broadinstitute/hellbender/tools/copynumber/ModelSegments.java]: Will add some links in my documentation PR (which is forthcoming).

File: src/main/java/org/broadinstitute/hellbender/tools/copynumber/ModelSegments.java [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+                    "The likelihood for an allelic count to be generated from a homozygous site will be integrated " +
+                    "from zero base-error rate up to this value.  Decreasing this value will increase " +
+                    "the number of sites assumed to be heterozygous for modeling.",
+            fullName = GENOTYPING_BASE_ERROR_RATE_LONG_NAME,
+            shortName = GENOTYPING_BASE_ERROR_RATE_SHORT_NAME,
+            optional = true
+    )
+    private double genotypingBaseErrorRate = 5E-2;
+
+    @Argument(
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply src/main/java/org/broadinstitute/hellbender/tools/copynumber/ModelSegments.java]: Will do this in my documentation PR (which is forthcoming).

File: src/main/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervals.java [hunk 6]
```diff
@@ -30,7 +30,8 @@
  * IntervalArgumentCollection. However, we encourage using only the -P flag.
  *
  * <p>The user can also specify the length of the bins (in bp) using the -BL option. If this is not commensurate with
- * the length of the padded intervals, then the last bin will be of different length than the others. </p>
+ * the length of the padded intervals, then the last bin will be of different length than the others.  If zero is
+ * specified, then no binning will be performed.</p>
  *
  * <p> The -O argument specifies a filename for the output bins, stored as a Picard interval list. </p>
  *
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply src/main/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervals.java]: Also punting to docs PR.

File: src/main/resources/org/broadinstitute/hellbender/tools/coveragemodel/log_norm_table.tsv [hunk 7]
```diff
@@ -1,160 +0,0 @@
--313.418938533	-313.213945211	-312.6005796	-311.5836531	-310.171080323	-308.373726483	-306.205201447	-303.681607623	-300.8212502	-297.644319153	-294.172552829	-290.428892909	-286.437140061	-282.221618791	-277.806858902	-273.217299699	-268.477021684	-263.609509109	-258.637445385	-253.582542132	-248.46540157	-243.305411019	-238.120667579	-232.927930521	-227.742598531	-222.578708793	-217.448954769	-212.364719635	-207.336122414	-202.372074084	-197.480341148	-192.667614428	-187.939581148	-183.300998603	-178.755768019	-174.307007442	-169.95712272	-165.707875857	-161.56045021	-157.515512123
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply src/main/resources/org/broadinstitute/hellbender/tools/coveragemodel/log_norm_table.tsv]: Test files for gCNV Spark that are no longer used.

File: scripts/cnv_wdl/somatic/cnv_somatic_panel_workflow.wdl [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+#    normal_bam_1    bam_idx_1
+#    normal_bam_2    bam_idx_2
+#    ...
+#
+# - The interval-list file is required for both WGS and WES workflows and should be a Picard or GATK-style interval list.
+#   These intervals will be padded on both sides by the amount specified by PreprocessIntervals.padding (default 250)
+#   and split into bins of length specified by PreprocessIntervals.bin_length (default 1000; specify 0 to skip binning).
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply scripts/cnv_wdl/somatic/cnv_somatic_panel_workflow.wdl]: Done here and in identical text in pair WDL.

**Pull request merged**